lib/java_buildpack/framework/jrebel_agent.rb

```diff
@@ -24,36 +24,51 @@ module Framework
     # Encapsulates the functionality for enabling zero-touch JRebel support.
     class JrebelAgent < JavaBuildpack::Component::VersionedDependencyComponent
 
+      def initialize(context, &version_validator)
+        super(context, &version_validator)
+        @component_name = 'JRebel Agent'
+      end
+
       # (see JavaBuildpack::Component::BaseComponent#compile)
       def compile
-        download_zip false
-        FileUtils.mv(download_location + 'jrebel.jar', @droplet.sandbox + jar_name)
-        FileUtils.remove_dir(download_location, true)
+        download_zip(true, @droplet.sandbox + 'install')
       end
 
       # (see JavaBuildpack::Component::BaseComponent#release)
       def release
         @droplet.java_opts
-          .add_javaagent(@droplet.sandbox + jar_name)
-          .add_bootclasspath_p(@droplet.sandbox + jar_name)
+          .add_agentpath(@droplet.sandbox + ('install/lib/' + lib_name))
           .add_system_property('rebel.remoting_plugin', true)
+          .add_system_property('rebel.log', true)
+          .add_system_property('rebel.base', @droplet.sandbox + '.jrebel')
+          .add_system_property('rebel.cloud.platform', 'cloudfoundry/java-buildpack')
       end
 
       protected
 
       # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
       def supports?
-        jrebel_configured?(@application.root) || jrebel_configured?(@application.root + 'WEB-INF/classes')
+        jrebel_configured?(@application.root) || jrebel_configured?(@application.root + 'WEB-INF/classes') ||
+          jars_with_jrebel_configured?(@application.root)
       end
 
       private
 
       def jrebel_configured?(root_path)
-        (root_path + 'rebel.xml').exist? && (root_path + 'rebel-remote.xml').exist?
+        (root_path + 'rebel-remote.xml').exist?
+      end
+
+      def jars_with_jrebel_configured?(root_path)
+        jars = (root_path + '**/*.jar').glob
+        jars.each do |jar|
+          output = `unzip -l "#{jar}" | grep "rebel-remote\\.xml$"`.strip
+          return true unless output.empty?
+        end
+        false
       end
 
-      def download_location
-        @droplet.sandbox + 'jrebel'
+      def lib_name
+        @configuration['architecture'] == '64-bit' ? 'libjrebel64.so' : 'libjrebel32.so'
```

Comment:
Rather than hardcoding the architecture in the configuration file, can you please determine it dynamically like we [do for the repository stuff](https://github.com/cloudfoundry/java-buildpack/blob/master/lib/java_buildpack/repository/repository_index.rb#L65-L67)?